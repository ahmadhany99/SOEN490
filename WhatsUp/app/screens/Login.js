/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { useState } from "react";
import { StyleSheet, Text, View, Image, Alert } from "react-native";
import AppButton from "../components/AppButton";
import AppTextInput from "../components/AppTextInput";
import Links from "../components/Links";
import Screen from "../components/Screen";
import TitleHeaders from "../components/TitleHeaders";
import logo from "../Images/w3.png";
import { auth, db } from "../firebase";
import { collection, getDocs, query, where } from "firebase/firestore";
import { signInWithEmailAndPassword } from "firebase/auth";
import { useNavigation } from "@react-navigation/native";
import colors from "../config/colors";
import { SimpleLineIcons } from "@expo/vector-icons";

export default function Login() {
  const navigation = useNavigation();
  const [email, setEmail] = useState("jasmine@organizer.com");
  const [password, setPassword] = useState("capstone123");
  const [isOrganizer] = useState(false);
  const handleLogin = async () => {
    await signInWithEmailAndPassword(auth, email, password)
      .then(async () => {
        const q = query(collection(db, "users"), where("email", "==", email));
        const querySnapshot = await getDocs(q);
        if (querySnapshot != null) {
          querySnapshot.forEach((doc) => {
            if (isOrganizer == doc.data().isOrganizer) {
              navigation.navigate("Attendee");
              setPassword("");
            } else {
              navigation.navigate("Organizer");
              setPassword("");
            }
          });
        }
      })
      .catch(() => {
        Alert.alert("Try again", "Invalid email or password.");
      });
  };
  const SignupPressed = async () => {
    navigation.navigate("SignUp");
  };

  const ResetPassword = () => {
    navigation.navigate("ResetPassword");
  };
  return (
    <Screen
      resizeMode='cover'
      backgroundImage='https://cdn.dribbble.com/users/479289/screenshots/4521207/media/c3e3bd246bbff91101a54e69daa8b1f0.gif'
      style={{ paddingHorizontal: "10%" }}
    >
      <Image
        source={logo}
        style={{ marginTop: 18, width: 166, height: 212, alignSelf: "center" }}
      />
      <TitleHeaders
        style={{ fontSize: 38, marginTop: 20, marginBottom: 5 }}
        title='Login'
        isTitle={true}
      />
      <AppTextInput
        placeholder='Email'
        keyboardType='email-address'
        onChangeText={(text) => setEmail(text.toLowerCase())}
        value={email}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <AppTextInput
        placeholder='Password'
        secureTextEntry
        onChangeText={(text) => setPassword(text)}
        value={password}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <View style={styles.organizertwo}>
        <Links
          style={styles.link}
          link='Forgot Password?'
          onPress={ResetPassword}
        />
      </View>
      <AppButton title='Login' onPress={handleLogin} />
      <View style={{ flexDirection: "row", justifyContent: "center" }}>
        <Text style={styles.text}>Don't have an account? </Text>
        <Links style={styles.link} link='Sign up' onPress={SignupPressed} />
      </View>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          paddingVertical: 15,
        }}
      >
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
        <View>
          <Text
            style={{
              width: 105,
              textAlign: "center",
              color: "darkgray",
              fontSize: 14,
            }}
          >
            Or login with
          </Text>
        </View>
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
      </View>
      <View style={styles.rowContainer}>
        <SimpleLineIcons
          name='social-facebook'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            alignSelf: "center",
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='facebook'
        />
        <SimpleLineIcons
          name='social-twitter'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingHorizontal: 40,
            alignSelf: "center",
          }}
          onPress={() => console.log("Facebook")}
          testID='twitter'
        />
        <SimpleLineIcons
          name='social-google'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='google'
        />
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  organizer: {
    alignItems: "flex-start",
    width: "50%",
  },
  organizertwo: {
    alignItems: "flex-end",
    width: "98%",
  },
  rowContainer: {
    flexDirection: "row",
    alignSelf: "center",
  },
  roundshape: {
    backgroundColor: "grey",
    height: 65,
    width: 65,
    justifyContent: "center",
    alignSelf: "center",
    alignContent: "center",
    borderRadius: 45,
  },
  text: {
    color: colors.darkerGrey,
    fontSize: 16,
  },
  text2: {
    fontSize: 16,
    color: colors.darkGrey,
  },
});
