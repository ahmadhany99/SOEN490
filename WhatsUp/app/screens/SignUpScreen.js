import React, { useState } from "react";
import { StyleSheet, Text, View, Button, Image, Alert } from "react-native";
import AppButton from "../components/AppButton";
import AppTextInput from "../components/AppTextInput";
import Links from "../components/Links";
import Screen from "../components/Screen";
import ScreenSubtitle from "../components/ScreenSubtitle";
import ScreenTitle from "../components/ScreenTitle";
import Checkbox from 'expo-checkbox';
import { auth, db, createUserWithEmailAndPassword, collection, addDoc } from "../firebase";
import { useNavigation } from "@react-navigation/native";
import colors from "../config/colors";
import BottomImg from "../components/BottomImg";
import AwesomeAlert from 'react-native-awesome-alerts';


function SignUpScreen() {
  
  const navigation = useNavigation();
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [isOrganizer, setIsOrganizer] = useState(false);
  const [valid, setValid] = useState(true);
  const [sAlert, setSAlert] = useState(false);
  
  
  const handleSignUp = async (firstName, lastName, email, password, isOrganizer) => {
    if (firstName.length == 0) {
      Alert.alert("Error","Please fill out your first name.")
      return;
    }
    if (lastName.length == 0) {
      Alert.alert("Error","Please fill out your last name.")
      return;
    }
    await createUserWithEmailAndPassword(auth, email, password)
      .then(userCredentials => {
          const user = userCredentials.user;
          console.log(user.email);
          Alert.alert('Account Created Succesfully');
          navigation.navigate('Login')
        })
        .then(() => {addDoc(collection(db, "users"), {
          firstName: firstName,
          lastName: lastName,
          email: email,
          isOrganizer: isOrganizer,
        }).catch((error) =>  handleErrorMsg(error))
      }).catch((error) => handleErrorMsg(error))
  }
  
  const handleErrorMsg = (error) => {
    if (error.message.includes('invalid-email')) {
      Alert.alert('Error','Invalid Email');
    }
    else if(error.message.includes('email-already-in-use')) {
      Alert.alert('Error','Email already in use');
    }
    else if(error.message.includes('internal-error')) {
      Alert.alert('Error','Please fill in a password');
    }
    console.log(error.message);
  }
  const handleConfirmPass = (confirmPassword) => {
    
    if (confirmPassword !== password) {
      setValid(false)
    } else {
      setValid(true)
    }
  }  

  return (
    <Screen style={{padding: 20, marginTop: 20}}>
      <ScreenTitle title="Sign Up" />
      <ScreenSubtitle subtitle="Please fill the following information" />
      <AppTextInput placeholder="First Name" onChangeText={text => setFirstName(text)}/>
      <AppTextInput placeholder="Last Name" onChangeText={text => setLastName(text)}/>
      <AppTextInput placeholder="Email" onChangeText={text => setEmail(text)} keyboardType='email-address'/>
      <AppTextInput placeholder="Password" onChangeText={text => setPassword(text)} secureTextEntry/>
      <AppTextInput 
        placeholder="Confirm Password" 
        onChangeText={text => handleConfirmPass(text)} 
        secureTextEntry
        style={!valid ? styles.invalidInput : styles.validInput}
        />
      <View style={{flexDirection: 'row', paddingTop: 16}}>
        <View style={styles.organizer}>
            <Text style={styles.text}>Are you an organizer?</Text>
        </View>
        <View style={styles.organizertwo}>
            <Checkbox style={styles.check} value={isOrganizer} onValueChange={setIsOrganizer}/>
        </View>
      </View>
      <AppButton
        style={{
          shadowColor: 'black', // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 3, //IOS
          elevation: 4, // Android
        }}
        title="Sign Up" 
        disabled={!valid} 
        onPress={() => handleSignUp(firstName, lastName, email, password, isOrganizer)}
      />  
        <View style={{flexDirection: 'row', justifyContent: 'center'}}>
          <Text style={styles.text}>Already have an account? </Text>
          <Links style={styles.link} link="Login" onPress={() => navigation.navigate('Login')} />
        </View>
        
        <BottomImg
        resizeMode = 'contain'
        />

    </Screen>
  );
}

const styles = StyleSheet.create({
    organizer: {
        alignItems: 'flex-start',
        width: '50%',
        color: colors.lightGrey,
    },
    organizertwo: {
      alignItems: 'flex-end',
      width: '50%',
    },
    check: {
      borderRadius: 6,
      borderWidth: 1,
    },
    validInput: {
      borderColor: colors.lightGrey,
    },
    invalidInput: {
      borderColor: colors.secondary,
    },
    text: {
      color: colors.darkerGrey,
      fontSize: 16,
    }
});

export default SignUpScreen;
