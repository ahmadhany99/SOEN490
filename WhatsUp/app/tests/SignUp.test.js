import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import { Alert } from "react-native";
import Login from "../screens/Login";
import SignUpScreen from "../screens/SignUpScreen";

jest.useFakeTimers();
it("renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <SignUpScreen />
    </NavigationContainer>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Successful Sign Up", async () => {
  render(
    <NavigationContainer>
      <SignUpScreen />
    </NavigationContainer>
  );
  await waitFor(() =>
    fireEvent.changeText(
      screen.getByPlaceholderText("First Name"),
      "Mohona"
    )
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Last Name"), 
    "Mazumdar"
    )
  );
  await waitFor(() => 
    fireEvent.press(screen.getByPlaceholderText("Email"),
    "mohona6646@hotmail.com"
    )
    );
  await waitFor(() => 
    fireEvent.press(screen.getByPlaceholderText("Password"),
    "capstone123"
    )
    );   
  await waitFor(() => 
    fireEvent.press(screen.getByPlaceholderText("Confirm Password"),
    "capstone123"
    )
    );  
    
    await waitFor(() => {
    fireEvent.press(screen.getAllByText("Sign Up")[1]);
      }); 

 
//   jest.spyOn(Alert, "alert");
//   await waitFor(() => {
//     fireEvent.press(screen.getAllByText("Sign Up")[1]);
//     expect(Alert.alert).toHaveBeenCalledWith("Email already in use");
//   });
});

test("Wrong Confirm Password", async () => {
    render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );
    await waitFor(() =>
      fireEvent.changeText(
        screen.getByPlaceholderText("First Name"),
        "Mohona"
      )
    );
    await waitFor(() =>
      fireEvent.changeText(screen.getByPlaceholderText("Last Name"), 
      "Mazumdar"
      )
    );
    await waitFor(() => 
      fireEvent.press(screen.getByPlaceholderText("Email"),
      "mohona6646@hotmail.com"
      )
      );
    await waitFor(() => 
      fireEvent.press(screen.getByPlaceholderText("Password"),
      "capstone123"
      )
      );   
    await waitFor(() => 
      fireEvent.press(screen.getByPlaceholderText("Confirm Password"),
      "capstone122"
      )
      );  
      
      const onClick = jest.fn();
      fireEvent.press(screen.getAllByText("Sign Up")[1]);
      expect(onClick).not.toHaveBeenCalled();
  
   
  //   jest.spyOn(Alert, "alert");
  //   await waitFor(() => {
  //     fireEvent.press(screen.getAllByText("Sign Up")[1]);
  //     expect(Alert.alert).toHaveBeenCalledWith("Email already in use");
  //   });
  });


  test("Successful navigation to login page", async () => {
    render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );

  await waitFor(() => fireEvent.press(screen.getByText("Login")));
  const ls = (
    <NavigationContainer>
      <Login />
    </NavigationContainer>
  );
  render(ls);
  expect(screen.getByText("Please enter your details"));
});

test("SetValid is False with Wrong Confirm Password", async () => {
    render(
      <NavigationContainer>
        <SignUpScreen />
      </NavigationContainer>
    );

    await waitFor(() => 
    fireEvent.press(screen.getByPlaceholderText("Password"),
    "capstone123"
    )
    );   

  await waitFor(() => 
    fireEvent.press(screen.getByPlaceholderText("Confirm Password"),
    "capstone122"
    )
    );  
    //expect(valid).toBe("false");
});


