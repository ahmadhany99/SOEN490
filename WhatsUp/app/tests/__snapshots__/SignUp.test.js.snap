// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 20,
        "padding": 20,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
        "marginLeft": "1%",
      }
    }
  >
    Sign Up
  </Text>
  <Text
    style={
      Object {
        "color": "#747474",
        "marginBottom": 6,
        "marginLeft": "1%",
        "paddingVertical": 6,
      }
    }
  >
    Please fill the following information
  </Text>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="First Name"
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Last Name"
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        undefined,
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#989797",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "1%",
          "marginVertical": 10,
          "paddingHorizontal": 12,
          "paddingVertical": 7,
          "width": "98%",
        },
        Object {
          "borderColor": "#989797",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Confirm Password"
      secureTextEntry={true}
      style={
        Object {
          "color": "black",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#989797",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#484545",
            "fontSize": 16,
          }
        }
      >
        Are you an organizer?
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
            "disabled": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#657786",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            },
            Object {
              "borderRadius": 6,
              "borderWidth": 1,
            },
            false,
            false,
            undefined,
            false,
          ]
        }
      />
    </View>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00c0a4",
        "borderRadius": 7,
        "elevation": 4,
        "justifyContent": "center",
        "marginLeft": "1%",
        "marginVertical": 40,
        "opacity": 1,
        "padding": 12,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
        "width": "98%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Helvetica Neue",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#484545",
          "fontSize": 16,
        }
      }
    >
      Already have an account? 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#00c0a4",
            "fontSize": 16,
            "fontWeight": "400",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "bottom": 0,
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "bottom": 0,
          "height": 345.82829504232166,
          "position": "absolute",
          "width": 750,
        },
      ]
    }
  />
</View>
`;
